src/osd/osd_types.cc

```diff
@@ -3761,7 +3761,7 @@ void pg_create_t::generate_test_instances(list<pg_create_t*>& o)
 
 void pg_hit_set_info_t::encode(bufferlist& bl) const
 {
-  ENCODE_START(1, 1, bl);
+  ENCODE_START(2, 2, bl);
   ::encode(begin, bl);
   ::encode(end, bl);
   ::encode(version, bl);
```

Comment:
You need to explicitly encode the using_gmt param here.  If you decode a pg_hit_set_info_t from an old osd and reencode it, it will appear to have using_gmt as true otherwise.